compiler/kernel/expr.go

```diff
@@ -100,11 +100,32 @@ func compileExpr(zctx *zed.Context, scope *Scope, e dag.Expr) (expr.Evaluator, e
 		return compileSetExpr(zctx, scope, e)
 	case *dag.MapExpr:
 		return compileMapExpr(zctx, scope, e)
+	case *dag.Agg:
+		return compileAggExpr(zctx, scope, e)
 	default:
 		return nil, fmt.Errorf("invalid expression type %T", e)
 	}
 }
 
+func compileAggExpr(zctx *zed.Context, scope *Scope, e *dag.Agg) (expr.Evaluator, error) {
```

Comment:
Nits: Call this compileAgg since it compiles a dag.Agg, not a dag.AggExpr. And rename e to a for a similar reason.